Sources/TuistKit/Services/Cloud/CloudAuthService.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+import TuistCloud
+import TuistCore
+import TuistLoader
+import TuistSupport
+
+protocol CloudAuthServicing: AnyObject {
+    /// It reads the cloud URL from the project's Config.swift and
+    /// authenticates the user on that server storing the credentials
+    /// locally on the keychain
+    func authenticate() throws
+}
+
+enum CloudAuthServiceError: FatalError, Equatable {
+    case missingCloudURL
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case .missingCloudURL:
+            return "The cloudURL attribute is missing in your project's configuration."
+        }
+    }
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .missingCloudURL:
+            return .abort
+        }
+    }
+}
+
+final class CloudAuthService: CloudAuthServicing {
+    /// Cloud session controller.
+    let cloudSessionController: CloudSessionControlling
+
+    /// Generator model loader.
+    let generatorModelLoader: GeneratorModelLoading
+
+    // MARK: - Init
+
+    convenience init() {
+        let manifetLoader = ManifestLoader()
+        let manifestLinter = ManifestLinter()
+        let generatorModelLoader = GeneratorModelLoader(manifestLoader: manifetLoader,
+                                                        manifestLinter: manifestLinter)
+        self.init(cloudSessionController: CloudSessionController(),
+                  generatorModelLoader: generatorModelLoader)
+    }
+
+    init(cloudSessionController: CloudSessionControlling,
+         generatorModelLoader: GeneratorModelLoading) {
+        self.cloudSessionController = cloudSessionController
+        self.generatorModelLoader = generatorModelLoader
+    }
+
+    // MARK: - CloudAuthServicing
+
+    func authenticate() throws {
+        let path = FileHandler.shared.currentPath
```

Comment:
does this mean that the commands only work if you run them from the same path were the Config file is?

Reply:
Not really. The config loader traverses the up the folders hierarchy until it finds a `Tuist/Config.swift` file.

Reply:
ok that's cool, I just didn't get why we're using `currentPath` as the path